lib/record_store/provider/ns1/client.rb

```diff
@@ -1,46 +1,58 @@
+require 'net/http'
 require 'ns1'
 
 module RecordStore
   class Provider::NS1 < Provider
-    class Error < StandardError; end
-
     class Client < ::NS1::Client
       def initialize(api_key:)
         super(api_key)
       end
 
       def zones
-        super
+        zones = super
+        raise_error(zones) if zones.is_a?(NS1::Response::Error)
+        zones
       end
 
       def zone(name)
-        super(name)
+        zone = super(name)
+        raise_error(zone) if zone.is_a?(NS1::Response::Error)
+        zone
       end
 
       def record(zone:, fqdn:, type:, must_exist: false)
         result = super(zone, fqdn, type)
-        raise(Error, result.to_s) if must_exist && result.is_a?(NS1::Response::Error)
+        raise_error(result) if must_exist && result.is_a?(NS1::Response::Error)
         return nil if result.is_a?(NS1::Response::Error)
```

Comment:
We can avoid checking twice the same thing here:

```suggestion
        if result.is_a?(NS1::Response::Error)
          raise_error(result) if must_exist
          return nil
        end
```